packages/flutter_tools/lib/src/macos/migrations/secure_restorable_state_migration.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../xcode_project.dart';
+
+const String _appDelegateFileBefore = r'''
+class AppDelegate: FlutterAppDelegate {
+  override func applicationShouldTerminateAfterLastWindowClosed(_ sender: NSApplication) -> Bool {
+    return true
+  }
+}''';
```

Comment:
We won't be able to migrate apps that modified their `AppDelegate`.

Do you think we should log a message recommending the developer overrides  `applicationSupportsSecureRestorableState` if:

1. We find the app delegate file
2. We can't find the string `applicationSupportsSecureRestorableState` in the app delegate file
3. We can't migrate the app delegate file due to modifications

Alternatively, should the tool catch the `Secure coding is automatically enabled for restorable state! However, ...` error and recommend adding `applicationSupportsSecureRestorableState` to the app delegate?

I don't feel strongly about these, feel free to skip.

Reply:
I thought about that a bit. My worry is that this log message gets output on every build, so for someone who, for whatever reason really doesn't want to override this, they'll be getting this warning on every build forever.

Perhaps that's no worse than getting the "You didn't override this method" on every single run, forever, from Apple?

Reply:
After discussing offline, I'll land this and we can always revert if people complain about it.